File: src/film/feature/feature.cpp
#include "feature.h"
#include "../../utils/io/io.h"
#include "../../utils/random/random.h"
#include <cstdio>

Film *Feature::In(const char *title, int year, FILE *input) {
  return new Feature(title, year, ReadString(input));
}

Film *Feature::InStochastic() {
  return new Feature(
      RandomString(RandomInt(1, 51)),
      RandomInt(1970, 3022),
      RandomString(RandomInt(1, 51)));
}

Feature::Feature(const char *title, int year, const char *director) : Film(title, year), director(director) {}

void Feature::Out(FILE *output) {
  fprintf(output, "Feature{title=%s,year=%d,director=%s}", title, year, director);
}

File: src/film/feature/feature.py
from typing import IO

from src.film.film import Film
from src.utils import io, rand
from src.utils.file_wrapper import FileWrapper


class Feature(Film):
    def __init__(self):
        super().__init__()
        self.director = "Unknown director"

    def read_from_file(self, file: FileWrapper):
        super().read_from_file(file)
        self.director = io.read_str(file, "Director")

    def fill_stochastic(self):
        super().fill_stochastic()
        self.director = rand.random_str()

    def write_to_file(self, file: IO):
        file.write('Feature{' + F'title={self.title},year={self.year},director={self.director}' + '}')
